src/plugins/orangehrmAdminPlugin/Dao/LocalizationDao.php

```diff
@@ -83,4 +87,79 @@ public function saveI18NLanguage(I18NLanguage $i18NLanguage): I18NLanguage
         $this->persist($i18NLanguage);
         return $i18NLanguage;
     }
+
+    /**
+     * @param I18NTranslationSearchFilterParams $i18NTargetLangStringSearchFilterParams
+     * @return array e.g. [0 => ['id' => 1, 'source' => 'About', 'note' => null, 'target' => '关于']]
+     */
+    public function getNormalizedTranslations(
+        I18NTranslationSearchFilterParams $i18NTargetLangStringSearchFilterParams
+    ): array {
+        $q = $this->getTranslationsQueryBuilderWrapper($i18NTargetLangStringSearchFilterParams)->getQueryBuilder();
+        $q->select(
+            'translation.id',
+            'langString.value AS source',
+            'langString.note AS note',
+            'translation.value AS target',
+        );
+        return $q->getQuery()->execute();
+    }
+
+    /**
+     * @param I18NTranslationSearchFilterParams $i18NTargetLangStringSearchFilterParams
+     * @return QueryBuilderWrapper
+     */
+    private function getTranslationsQueryBuilderWrapper(
+        I18NTranslationSearchFilterParams $i18NTargetLangStringSearchFilterParams
+    ): QueryBuilderWrapper {
+        $q = $this->createQueryBuilder(I18NLangString::class, 'langString');
+        $q->leftJoin(
+            'langString.translations',
+            'translation',
+            Expr\Join::WITH,
+            'IDENTITY(translation.language) = :langId'
+        );
+        $q->setParameter('langId', $i18NTargetLangStringSearchFilterParams->getLanguageId());
+
+        if (!is_null($i18NTargetLangStringSearchFilterParams->getOnlyTranslated())) {
+            if ($i18NTargetLangStringSearchFilterParams->getOnlyTranslated() === true) {
+                $q->andWhere($q->expr()->isNotNull('translation.translated'));
+            } elseif ($i18NTargetLangStringSearchFilterParams->getOnlyTranslated() === false) {
+                $q->andWhere($q->expr()->isNull('translation.translated'));
+            }
+        }
+
+        if (!empty($i18NTargetLangStringSearchFilterParams->getGroupId())) {
+            $q->andWhere('langString.group = :groupId')
+                ->setParameter('groupId', $i18NTargetLangStringSearchFilterParams->getGroupId());
+        }
+
+        if (!empty($i18NTargetLangStringSearchFilterParams->getSourceText())) {
+            $q->andWhere($q->expr()->like('langString.value', ':sourceText'))
+                ->setParameter('sourceText', '%' . $i18NTargetLangStringSearchFilterParams->getSourceText() . '%');
+        }
+
+        if (!empty($i18NTargetLangStringSearchFilterParams->getTranslatedText())) {
+            $q->andWhere($q->expr()->like('translation.value', ':translatedText'))
+                ->setParameter(
+                    'translatedText',
+                    '%' . $i18NTargetLangStringSearchFilterParams->getTranslatedText() . '%'
+                );
+        }
```

Comment:
is there a specific reason for using `empty` rather than `is_null`? are you checking for empty strings also?